jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ConnectionSettingsMenuBuilder.kt

```diff
@@ -0,0 +1,193 @@
+            otherPartitions.forEach {
+                val partitionGroup = DefaultActionGroup.createPopupGroup { it.displayName }
+                partitionGroup.addAll(createRegionGroupActions(it.regions, currentSelection = null, onChange))
+
+                otherPartitionActionGroup.add(partitionGroup)
+            }
+
+            add(Separator.create())
+            add(otherPartitionActionGroup)
+        }
+    }
+
+    private fun createRegionGroupActions(regions: Collection<AwsRegion>, currentSelection: AwsRegion?, onChange: (AwsRegion) -> Unit) = buildList<AnAction> {
+        regions.groupBy { it.category }
+            .forEach { (category, categoryRegions) ->
+                add(Separator.create(category))
+                categoryRegions.sortedBy { it.displayName }
+                    .forEach { add(SwitchRegionAction(it, it == currentSelection, onChange)) }
+            }
+    }
+
+    private fun createProfileActions(): List<AnAction> = buildList {
+        val (currentSelection, onChange) = credentialsSelectionSettings ?: return@buildList
+
+        add(Separator.create(message("settings.credentials")))
+
+        val credentialManager = CredentialManager.getInstance()
+        credentialManager.getCredentialIdentifiers().forEach {
+            add(SwitchCredentialsAction(it, it == currentSelection, onChange))
+        }
+
+        add(Separator.create())
+        add(ActionManager.getInstance().getAction("aws.settings.upsertCredentials"))
+    }
+
+    // Helper actions, note: these are public to help make tests easier by leveraging instanceOf checks
+
+    abstract inner class DumbAwareToggleAction<T>(
+        title: String,
+        val value: T,
+        private val selected: Boolean,
+        private val onSelect: (T) -> Unit
+    ) : ToggleAction(title), DumbAware {
+        override fun isSelected(e: AnActionEvent): Boolean = selected
+
+        override fun setSelected(e: AnActionEvent, state: Boolean) {
+            if (!isSelected(e)) {
+                onSelect.invoke(value)
+            }
+        }
+    }
+
+    inner class SwitchRegionAction(
+        value: AwsRegion,
+        selected: Boolean,
+        onSelect: (AwsRegion) -> Unit
+    ) : DumbAwareToggleAction<AwsRegion>(value.displayName, value, selected, onSelect)
+
+    inner class SwitchCredentialsAction(
+        value: CredentialIdentifier,
+        selected: Boolean,
+        onSelect: (CredentialIdentifier) -> Unit
+    ) : DumbAwareToggleAction<CredentialIdentifier>(value.displayName, value, selected, onSelect)
+
+    companion object {
+        private const val ID = "ConnectionSettingsMenu"
+
+        fun connectionSettingsMenuBuilder(): ConnectionSettingsMenuBuilder = ConnectionSettingsMenuBuilder()
+
+        fun projectConnectionSettingsMenu(project: Project, dataContext: DataContext, showRegions: Boolean = true, showCredentials: Boolean = true): ListPopup {
```

Comment:
feels like we could use ProjectLevelSettingSelector here?

Reply:
Good point, it predates me building ProjectLevelSettingSelector. Moved to it.